### DISCUSSION TITLE
Inject object in query

### DISCUSSION THREAD

**Discussion by untel4444** [symfony/symfony#49279] [Q&A] [answered]
Hello,
Is it possible to directly inject object for querying throught a repository ?

For exemple : 
$inventaire = new Inventaire();
$inventaire ->setCategorie(1);
$inventaire ->setColor(2);

//search will return all objects meeting properties - not using findByX
$inventaires = $inventaireRepository->findSearch($inventaire);

Thanks
Julien

**Comment by MatTheCat** [comment]
Hello,

I don’t understand your question: what is `findSearch` supposed to do?

**Comment by lyrixx** [comment, marked as answer]
This is not provided by default by doctrine.  But they have something called [criteria](https://www.doctrine-project.org/projects/doctrine-collections/en/stable/expression-builder.html#expression-builder). You can use this and / or build something on top of it 👍🏼 

Anyway, this is more a question about doctrine and not symfony, this means it's not the best place to ask for help.

**Comment by untel4444** [comment]
Hello MatTheCat,
A method returning all objects matching the properties of the object injected.

Lyrixx, you are surely right

**Comment by untel4444** [comment]
Thanks a lot Lyrixx,  I will look at this
